### PR TITLE
Add fsGroup to controller deployment

### LINKED ISSUE #890: controller won't run on k8s > 1.20 without fsGroup
I identified an issue while upgrading our (outdated) 1.17 cluster to 1.21.
With the TokenRequest and TokenRequestProjection featuregates graduated to GA in 1.20 the SerivceAccount token is now mounted via the [projected volume API](https://kubernetes.io/docs/concepts/storage/volumes/#projected).
Since metallb runs as nobody user and the projected token is only readable by root(600) or root and fsGroup if set(640) we need to set the fsGroup in the helm chart, or at least, make securityContext configurable via values.yaml.

I'll happily provide a PR for the chart, I'm just curious if I'm doing something wrong and there's another way to run metallb on k8s > 1.20 or if just nobody run it on 1.20+ yet (what I couldn't imagine tbh).

### PR DESCRIPTION
**Pull request by TheKangaroo**
Fixes #890
we need to set fsGroup for the controller to read the k8s api token.

### CODE REVIEW

**Comment by russellb** [comment]
@gclawes FYI on this PR.  I wonder if we should add a comment by the chart version to clarify that it gets updated as part of our release process and doesn't need to be updated in PRs?

**Comment by gclawes** [comment]
The description says this PR is adding to the values, but I only see `controller.yaml` changing.

**Comment by gclawes** [comment]
Edit: Ah, I see here: https://github.com/metallb/metallb/issues/890#issuecomment-862424413

@russellb given that this PR has been reduced in scope, want me to make the comment in another PR?

**Comment by russellb** [comment]
> @russellb given that this PR has been reduced in scope, want me to make the comment in another PR?

Yes, that'd be great

**Comment by russellb** [comment]
I'd like to add a release note about this change.  How about this, based on the bug report:

```
 -  Add `fsGroup` to the MetalLB controller deployment to address compatibility with Kubernetes 1.20 and later. With the TokenRequest and TokenRequestProjection featuregates graduated to GA in 1.20, the SerivceAccount token is now mounted via the projected volume API. The security context needed an update to ensure the controller is able to read the token.  (Issue #890)
```

**Comment by gclawes** [comment]
LGTM on the actual helm chart diff and the release notes.  Should the manifests be updated too?

**Comment by gclawes** [comment]
@russellb I will make the helm chart version note in #886

**Comment by champtar** [comment]
s/SerivceAccount/ServiceAccount/

File: charts/metallb/templates/controller.yaml [hunk 1]
```diff
@@ -36,6 +36,7 @@ spec:
       securityContext:
         runAsNonRoot: true
         runAsUser: 65534 # nobody
+        fsGroup: 65534
```
Comment (russellb) [inline charts/metallb/templates/controller.yaml:39]: This will need to go in manifests/metallb.yaml too, right?

**Comment by russellb** [comment]
> s/SerivceAccount/ServiceAccount/

thanks, fixed

**Comment by TheKangaroo** [comment]
Sorry for the confusion to comment in https://github.com/metallb/metallb/issues/890 instead of the PR before.
I think it would be best to default fsGroup to 65534, but add the option to override it via values.yaml.
For the release note, I dug a bit deeper in the kubernetes Docs and think `TokenRequest` and `TokenRequestProjection` are just part of the problem. The root cause for this is the changed ServiceAccount admission controller behaviour (https://github.com/kubernetes/kubernetes/issues/70679) which graduated to beta in 1.21.

File: charts/metallb/templates/controller.yaml (hunk 1 above)
Comment (TheKangaroo) [reply charts/metallb/templates/controller.yaml]: thanks, I fixed that

**Comment by russellb** [comment]
an updated release note ...

```
 -  Add `fsGroup` to the MetalLB controller deployment to address compatibility with Kubernetes 1.21 
    and later. See [Kubernetes issue #70679](https://github.com/kubernetes/kubernetes/issues/70679).
    This ensures the MetalLB controller can read the service account token volume.  (Issue #890)
```

Look OK to you, @TheKangaroo ?  or propose changes as you see fit

**Comment by TheKangaroo** [comment]
Looks good to me, thank you :)

**Comment by russellb** [comment]
also backported these changes to the v0.10 branch

**Pull request merged**